packages/path_provider/path_provider_android/lib/path_provider_android.dart

```diff
@@ -81,12 +81,20 @@ class PathProviderAndroid extends PathProviderPlatform {
   Future<List<String>?> getExternalStoragePaths({
     StorageDirectory? type,
   }) async {
-    return (await _api.getExternalStoragePaths(_convertStorageDirectory(type)))
-        .cast<String>();
+    return _getExternalStoragePaths(type: type);
   }
 
   @override
-  Future<String?> getDownloadsPath() {
-    throw UnsupportedError('getDownloadsPath is not supported on Android');
+  Future<String?> getDownloadsPath() async {
+    final List<String> paths =
+        await _getExternalStoragePaths(type: StorageDirectory.downloads);
+    return paths.isNotEmpty ? paths.first : null;
```

Comment:
Prefer conditions without negation when when using `?:` or if/else, to avoid the implicit double-negation for the "else" branch (i.e., currently this is "if paths is not empty, return the first path; if paths is not not empty, return null). So `return paths.isEmpty ? null : paths.first;`